### ISSUE TITLE
Installation issue: geant4@10.05.p01spack

### ISSUE THREAD

**Issue by thomasbrTTU** [spack/spack#15489]
### Spack version
```console
$ spack --version
0.13.3
```
### Steps to reproduce the issue
```console
$ spack install geant4
```

### Platform and user environment
```
$ uname -a 
Linux nocona-test.hpcc.ttu.edu 3.10.0-1062.9.1.el7.x86_64 #1 SMP Fri Dec 6 15:49:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -d
Description:    CentOS Linux release 7.7.1908 (Core)
``` 

### Additional information
```console
$ spack spec --install-status geant4

[root@nocona-test geant4]# spack spec --install-status geant4
Input spec
--------------------------------
 -   geant4

Concretized
--------------------------------
==> Warning: gcc@8.3.0 cannot build optimized binaries for "zen2". Using best target possible: "zen"
==> Error: An unsatisfiable variant constraint has been detected for spec:

    xerces-c@3.2.2%gcc@8.3.0 cxxstd=default transcoder=gnuiconv arch=linux-centos7-zen


while trying to concretize the partial spec:

    geant4@10.05.p01%gcc@8.3.0 build_type=RelWithDebInfo cxxstd=11 +data~motif~opengl~qt+threads~vecgeom~x11 arch=linux-centos7-zen
        ^cmake@3.16.2%gcc@8.3.0~doc+ncurses+openssl+ownlibs~qt arch=linux-centos7-zen
            ^ncurses
                ^pkgconfig
            ^openssl
                ^perl@5.14.0:
                    ^gdbm
                        ^readline
                ^zlib@1.2.11%gcc@8.3.0+optimize+pic+shared arch=linux-centos7-zen


geant4 requires xerces-c variant cxxstd=11, but spec asked for cxxstd=default
```

### General information

- [ X] I have run `spack --version` and reported the version of Spack
- [ X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by drbenmorgan** [comment]
Yep, this is an error due to the additional `depends_on(xerces-c)` here:

https://github.com/spack/spack/blob/05d0c7b4776ceb3afa5cebdab31dd42b494764bd/var/spack/repos/builtin/packages/geant4/package.py#L61

There's a PR (#15513) that addresses this, though you can patch locally by removing the above line.

**Comment by thomasbrTTU** [comment]
Thanks that works. Will someone change it for the Spack packages.

**Comment by drbenmorgan** [comment]
It's merged in `develop` with #15513, so will be in the next Spack release.
